### PR TITLE
SDK update for 0.2

### PR DESCRIPTION
**Pull request by MattPlumlee**
This change outlines the specific requirements for testing we are adopting for 0.2.  @wildsm , this was discussed on 07/11/2022.

Let's keep the conversation here unless we want to swap to slack.

### CODE REVIEW

**Comment by DanielRPhillips** [comment]
I have reviewed the changes and they look good to me.

**Review by odell — APPROVED** [review]
Looks good.

**Review by furnstahl — APPROVED** [review]
Looks good.

**Comment by ozgesurer** [comment]
I know my opinion doesn't matter, but since I have already told this Matt and my review is also requested, for the records I wanted to share it with you in response to this PR. 

As I indicated in the meeting I am not in favor of modifying the SDK document at this stage of staging stage. Given that me, Alexandra, Dan, Odell prepared the SDK files using the v0.1 and our ultimate goal for v1.0 is to provide a comprehensive test suite, then I don't see why we are moving this mandatory policy to down below as recommended (you can find all filled SDK docs in `develop` branch, Alexandra's is on the way). This SDK change does not change anything in our codes and tiny repos at this point, and instead of rewarding the tiny amount of coders within  BAND, this just creates an additional paper work for those who contribute at this stage.

**Review by wildsm — CHANGES_REQUESTED** [review]
I believe the far better practice would be to move what is newly labeled "R4" to be labeled "R14". This will make the changes clearer and mean that the R1-13 do not change numbers.

**Comment by wildsm** [comment]
> I know my opinion doesn't matter, but since I have already told this Matt and my review is also requested, for the records I wanted to share it with you in response to this PR.
> 
> As I indicated in the meeting I am not in favor of modifying the SDK document at this stage of staging stage. Given that me, Alexandra, Dan, Odell prepared the SDK files using the v0.1 and our ultimate goal for v1.0 is to provide a comprehensive test suite, then I don't see why we are moving this mandatory policy to down below as recommended (you can find all filled SDK docs in `develop` branch, Alexandra's is on the way). This SDK change does not change anything in our codes and tiny repos at this point, and instead of rewarding the tiny amount of coders within BAND, this just creates an additional paper work for those who contribute at this stage.

Your opinion matters critically @ozgesurer .

I believe that the delayed release, requested changes, and authorship discussion may have addressed some of your concerns. I believe additional issues and/or discussion should be added regarding testing. One concern is the problem with actions, which needs to be addressed to facilitate testing ([Issue 32](https://github.com/bandframework/bandframework/issues/32)).

Please update here.

**Review by ozgesurer — APPROVED** [review]

File: resources/sdkpolicies/bandsdk.md [hunk 1]
```diff
@@ -19,13 +19,14 @@ The BAND Framework will conform with BAND Software Development Kit (SDK) require
...
+| 2. | Provide a documented, reliable way to contact the development team.
+| 3. | Have README file in the top directory.
+| 4. | Within the README file have a specific set of testing procedures for a user to verify the software was installed and run correctly.
+| 5. | Come with an open-source license.
+| 6. | Provide a runtime API to return the current version number of the software.
+| 7. | Provide a BAND team-accessible repository.
+| 8. | Must allow installing, building, and linking against an outside copy of all imported software that is externally developed and maintained.
+| 9. | Have no hardwired print or IO statements that cannot be turned off.
 
 ### Recommended Policies
```
Comment (wildsm) [inline resources/sdkpolicies/bandsdk.md:30]: Here is what I think t should be:

| 2. | Have a README file in the top directory that states a specific set of testing procedures for a user to verify the software was installed and run correctly.
| 3. | Provide a documented, reliable way to contact the development team.
| 4. | Come with an open-source license.
| 5. | Provide a runtime API to return the current version number of the software.
| 6. | Provide a BAND team-accessible repository.
| 7. | Must allow installing, building, and linking against an outside copy of all imported software that is externally developed and maintained.
| 8. | Have no hardwired print or IO statements that cannot be turned off.

File: resources/sdkpolicies/bandsdk.md [hunk 2]
```diff
@@ -34,13 +35,14 @@ The BAND Framework will conform with BAND Software Development Kit (SDK) require
...
+|**R5.**| Document versions of packages that it works with or depends upon, preferably in machine-readable form.
+|**R6.**| Have SUPPORT, LICENSE, and CHANGELOG files in top directory.
+|**R7.**| Have sufficient documentation to support use and further development.
+|**R8.**| Be buildable using 64-bit pointers; 32-bit is optional.
+|**R9.**| Do not assume a full MPI communicator; allow for user-provided MPI communicator.
+|**R10.**| Use a limited and well-defined name space (e.g., symbol, macro, library, include).
+|**R11.**| Give best effort at portability to key architectures.
+|**R12.**| Install headers and libraries under `<prefix>/include` and `<prefix>/lib`, respectively.
+|**R13.**| All BAND compatibility changes should be sustainable.
+|**R14.**| Respect system resources and settings made by other previously called packages.
```
Comment (wildsm) [inline resources/sdkpolicies/bandsdk.md]: Here is what I think it should be:

|**R4.**| Document versions of packages that it works with or depends upon, preferably in machine-readable form.
|**R5.**| Have SUPPORT, LICENSE, and CHANGELOG files in top directory.
|**R6.**| Have sufficient documentation to support use and further development.
|**R7.**| Be buildable using 64-bit pointers; 32-bit is optional.
|**R8.**| Do not assume a full MPI communicator; allow for user-provided MPI communicator.
|**R9.**| Use a limited and well-defined name space (e.g., symbol, macro, library, include).
|**R10.**| Give best effort at portability to key architectures.
|**R11.**| Install headers and libraries under `<prefix>/include` and `<prefix>/lib`, respectively.
|**R12.**| All BAND compatibility changes should be sustainable.
|**R13.**| Respect system resources and settings made by other previously called packages.
|**R14.**| Provide a comprehensive, automated test suite for correctness of installation verification.

File: resources/sdkpolicies/bandsdk.md (hunk 1 above)
Comment (wildsm) [reply resources/sdkpolicies/bandsdk.md:30]: I have committed the above

**Review by wildsm — APPROVED** [review]
Good to me now, but will add another reviewerr

**Review by ozgesurer — APPROVED** [review]
Looks good.

**Pull request merged**